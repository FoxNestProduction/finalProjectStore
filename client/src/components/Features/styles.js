export const flexCenter = {
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'center',
};

export const featuresBG = {
  backgroundImage: {
    mobile: `url(${process.env.PUBLIC_URL}/img/FeaturesMobileBG.png)`,
    tablet: `url(${process.env.PUBLIC_URL}/img/FeaturesTabletBG.png)`,
  },
  backgroundSize: 'cover',
  width: '100%',
};

export const featuresContainer = {
  display: 'flex',
  flexDirection: {
    mobile: 'column',
    tablet: 'row',
  },
  alignItems: 'center',
  justifyContent: 'space-between',
  height: {
    mobile: '748px',
    tablet: '160px',
    desktop: '220px',
  },
  p: {
    mobile: '100px 0',
    tablet: '40px 0',
    desktop: '56px 0',
  },
  mb: {
    mobile: '67px',
    tablet: '85px',
    desktop: '115px',
  },
};

export const itemContainer = {
  width: '100%',
  flexDirection: 'column',
};

export const itemTitle = {
  fontSize: {
    mobile: '43px',
    tablet: '30px',
    desktop: '40px',
  },
  fontWeight: 700,
  color: 'text.primaryLight',
};

export const descriptionBox = {
  width: '100%',
  flexDirection: 'column',
  color: 'text.secondaryLight',
};

export const description = {
  fontSize: {
    mobile: '14px',
    tablet: '10px',
    desktop: '16px',
  },
};

export const lineBreak = {
  bgcolor: '#CBCBCB',
  opacity: '.15',
  width: {
    mobile: '240px',
    tablet: '1px',
  },
  height: {
    mobile: '1px',
    tablet: '80px',
    desktop: '110px',
  },
};
