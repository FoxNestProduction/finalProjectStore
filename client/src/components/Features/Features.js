import { Box, Container, Typography } from '@mui/material';
import React, { memo } from 'react';
import { flexCenter, featuresContainer, itemContainer, itemTitle, descriptionBox, description, lineBreak, featuresBG } from './styles';

const Features = () => {
  return (
    <Box sx={featuresBG}>
      <Container component="section" sx={featuresContainer}>
        <Box sx={{ ...flexCenter, ...itemContainer }}>
          <Typography component="h1" sx={itemTitle}>10K+</Typography>
          <Box sx={{ ...flexCenter, ...descriptionBox }}>
            <Typography component="p" sx={description}>Satisfied Costumers</Typography>
            <Typography component="p" sx={description}>All Great Over The World</Typography>
          </Box>
        </Box>
        <Typography component="span" sx={lineBreak} />
        <Box sx={{ ...flexCenter, ...itemContainer }}>
          <Typography component="h1" sx={itemTitle}>4M</Typography>
          <Box sx={{ ...flexCenter, ...descriptionBox }}>
            <Typography component="p" sx={description}>Healthy Dishes Sold</Typography>
            <Typography component="p" sx={description}>Including Milk Shakes Smooth</Typography>
          </Box>
        </Box>
        <Typography component="span" sx={lineBreak} />
        <Box sx={{ ...flexCenter, ...itemContainer }}>
          <Typography component="h1" sx={itemTitle}>99.99%</Typography>
          <Box sx={{ ...flexCenter, ...descriptionBox }}>
            <Typography component="p" sx={description}>Reliable Customer Support</Typography>
            <Typography component="p" sx={description}>We Provide Great Experiences</Typography>
          </Box>
        </Box>
      </Container>
    </Box>
  );
};

export default memo(Features);
