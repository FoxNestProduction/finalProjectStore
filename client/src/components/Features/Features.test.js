import React from "react";
import { render, fireEvent, screen } from "@testing-library/react";
import Features from "./Features";

describe('Features component', () => {
  test('should render Features component', () => {
    const { asFragment } = render(<Features />);
    expect(asFragment()).toMatchSnapshot();
  });

  // test('ScrollTop component should scroll back to top when clicked', () => {
  //   render(<ScrollTop />);

  //   const scrollButton = screen.getByLabelText('scroll back to top');

  //   fireEvent.click(scrollButton);
  
  //   expect(window.scrollY).toBe(0);
  // });
});