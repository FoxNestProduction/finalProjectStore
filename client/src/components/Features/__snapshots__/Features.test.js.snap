// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Features component should render Features component 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-wgdcnq"
  >
    <section
      class="MuiContainer-root MuiContainer-maxWidthLg css-u3wzei-MuiContainer-root"
    >
      <div
        class="MuiBox-root css-1lx9wp3"
      >
        <h1
          class="MuiTypography-root MuiTypography-body1 css-1t43y5d-MuiTypography-root"
        >
          10K+
        </h1>
        <div
          class="MuiBox-root css-o909xz"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-1oyztqb-MuiTypography-root"
          >
            Satisfied Costumers
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 css-1oyztqb-MuiTypography-root"
          >
            All Great Over The World
          </p>
        </div>
      </div>
      <span
        class="MuiTypography-root MuiTypography-body1 css-1yvp5mi-MuiTypography-root"
      />
      <div
        class="MuiBox-root css-1lx9wp3"
      >
        <h1
          class="MuiTypography-root MuiTypography-body1 css-1t43y5d-MuiTypography-root"
        >
          4M
        </h1>
        <div
          class="MuiBox-root css-o909xz"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-1oyztqb-MuiTypography-root"
          >
            Healthy Dishes Sold
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 css-1oyztqb-MuiTypography-root"
          >
            Including Milk Shakes Smooth
          </p>
        </div>
      </div>
      <span
        class="MuiTypography-root MuiTypography-body1 css-1yvp5mi-MuiTypography-root"
      />
      <div
        class="MuiBox-root css-1lx9wp3"
      >
        <h1
          class="MuiTypography-root MuiTypography-body1 css-1t43y5d-MuiTypography-root"
        >
          99.99%
        </h1>
        <div
          class="MuiBox-root css-o909xz"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-1oyztqb-MuiTypography-root"
          >
            Reliable Customer Support
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 css-1oyztqb-MuiTypography-root"
          >
            We Provide Great Experiences
          </p>
        </div>
      </div>
    </section>
  </div>
</DocumentFragment>
`;
